### ISSUE TITLE
some issues about pretrain

### ISSUE THREAD

**Issue by QY6616** [limirs/GeoExplorer#5]
In the GOMAA-Geo paper, you mention using a single A100 GPU and training for 24 hours. When I reproduced your code, I used an RTX 5090 (32 GB) for pre-training, but the actual GPU memory usage was only about 4 GB, and the whole pre-training process finished within a few hours. I also noticed that the number of parameters during the pre-training stage is 238M, which does not seem to match the full parameter size of a Falcon-7B model.

<img width="790" height="179" alt="Image" src="https://github.com/user-attachments/assets/a5bad4b0-e6ef-461b-9d30-e74ee5b2a914" />

**Comment by limirs** [comment]
Hi, we had the same experience when reproducing GOMAA-Geo. We used a single 4090. The pretraining was fast, and the GPU memory usage was low. However, to ensure a fair comparison with our baseline, we followed the same settings as GOMAA-Geo in GeoExplorer.

**Comment by QY6616** [comment]
Thank you very much for your reply. I have now successfully completed both the training and testing. However, the results on the MASA test set appear to differ from the numbers reported in your paper. I am wondering whether I might have overlooked some details. I have confirmed that my data split is correct, and when I tested using the weights provided by GOMAA-Geo, the results were very close to those reported in their paper. In addition, during the RL stage, is setting cfg.train.hparams.max_training_timesteps = int(1e8) too large? At what point did you stop the RL training in your implementation?
<img width="733" height="278" alt="Image" src="https://github.com/user-attachments/assets/136fa24c-896b-401d-bef4-2cf499b6e7bd" />

**Comment by QY6616** [comment]
@limirs I sincerely apologize for bothering you again. During training, I observed that as the number of exploration steps increases, the performance on the MASA test set does improve; however, there is still a noticeable gap compared to the results reported in the paper. I encountered the same issue when reproducing GOMAA-GEO.

Given that this problem appears to be common across both reproductions, I suspect that I may have overlooked some important details. I was wondering whether you might be willing to share any insights or practical experience from your training process. Thank you very much for your time and help.

<img width="319" height="35" alt="Image" src="https://github.com/user-attachments/assets/87a390f3-4cd3-41b4-a361-da235eeadbaa" />

**Comment by limirs** [comment]
Hi, thanks for letting us know. The RL training process may depend on many factors and can be unstable as well. Given the info you provided, I would suggest you try:

- Different pretraining weights. The default pretraining is 300 epochs; however, we noticed that different pretraining weights as starting points can affect RL convergence: you might need more time to train the model during the RL stage if the starting point is inappropriate. That's probably why you noticed that performance improves as training continues.

- Different random seeds. The default random seed in both GOMAA-Geo and GeoExplorer is 42. It works for our local environment, but perhaps not yours. Note that the random seed could affect the random trajectory generation and start-end pair generation, and the generation might be different across local environments, even with the same random seed. Given that the RL training is not very stable, it might make a difference in both pretraining and the RL training. We noticed that with certain random seeds, the RL training could get stuck. However, for most of those we tested, the results are comparable. 

Since we don't have more details, those suggestions could be completely irrelevant. If you are willing to share your training log and other information with us via email, we could probably have a closer look.

**Comment by Ziyang-X** [comment]
> Thank you very much for your reply. I have now successfully completed both the training and testing. However, the results on the MASA test set appear to differ from the numbers reported in your paper. I am wondering whether I might have overlooked some details. I have confirmed that my data split is correct, and when I tested using the weights provided by GOMAA-Geo, the results were very close to those reported in their paper. In addition, during the RL stage, is setting cfg.train.hparams.max_training_timesteps = int(1e8) too large? At what point did you stop the RL training in your implementation? <img alt="Image" width="733" height="278" src="https://private-user-images.githubusercontent.com/126632974/524701196-136fa24c-896b-401d-bef4-2cf499b6e7bd.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzI1MjEzODMsIm5iZiI6MTc3MjUyMTA4MywicGF0aCI6Ii8xMjY2MzI5NzQvNTI0NzAxMTk2LTEzNmZhMjRjLTg5NmItNDAxZC1iZWY0LTJjZjQ5OWI2ZTdiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMwM1QwNjU4MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZTlmOGUwMzJhNzA5ZWQyNGQ5ODk0OTBkMzVlM2Y2NjMzZjM3NGZhODg4NmYzYWQ1ZTI5MzRkM2Q4NjI1ZTJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.04rbNQkiplzy_B1gMYuU7AS6qbJDIQytWo8Amhw5xP8">

Hello, when I reproduce this paper, the success ratio is closed to this result. I have pretrained 300 epochs, and train over 6000 epochs. I wonder how did you improve the results, do you use params the author offered? 

This paper shows that "The Massachusetts Buildings (Masa) dataset [25] consists of 1188 high resolution images of the Boston area."
But I can only see 151 images in https://www.kaggle.com/datasets/balraj98/massachusetts-buildings-dataset (provided by GOMAA-Geo), and there are only 10 images for test, 4 images for val. If this is normal?

I am looking forward for your reply, thank you very much!!!!!!!!!!!!

**Comment by QY6616** [comment]
> > Thank you very much for your reply. I have now successfully completed both the training and testing. However, the results on the MASA test set appear to differ from the numbers reported in your paper. I am wondering whether I might have overlooked some details. I have confirmed that my data split is correct, and when I tested using the weights provided by GOMAA-Geo, the results were very close to those reported in their paper. In addition, during the RL stage, is setting cfg.train.hparams.max_training_timesteps = int(1e8) too large? At what point did you stop the RL training in your implementation? <img alt="Image" width="733" height="278" src="https://private-user-images.githubusercontent.com/126632974/524701196-136fa24c-896b-401d-bef4-2cf499b6e7bd.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzI1MjEzODMsIm5iZiI6MTc3MjUyMTA4MywicGF0aCI6Ii8xMjY2MzI5NzQvNTI0NzAxMTk2LTEzNmZhMjRjLTg5NmItNDAxZC1iZWY0LTJjZjQ5OWI2ZTdiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMwM1QwNjU4MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZTlmOGUwMzJhNzA5ZWQyNGQ5ODk0OTBkMzVlM2Y2NjMzZjM3NGZhODg4NmYzYWQ1ZTI5MzRkM2Q4NjI1ZTJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.04rbNQkiplzy_B1gMYuU7AS6qbJDIQytWo8Amhw5xP8">
> 
> Hello, when I reproduce this paper, the success ratio is closed to this result. I have pretrained 300 epochs, and train over 6000 epochs. I wonder how did you improve the results, do you use params the author offered?
> 
> This paper shows that "The Massachusetts Buildings (Masa) dataset [25] consists of 1188 high resolution images of the Boston area." But I can only see 151 images in https://www.kaggle.com/datasets/balraj98/massachusetts-buildings-dataset (provided by GOMAA-Geo), and there are only 10 images for test, 4 images for val. If this is normal?
> 
> I am looking forward for your reply, thank you very much!!!!!!!!!!!!
可以留一下你的联系方式，我们详细沟通
